### PR TITLE
Fix issue 26352: split up preserveAspectRatio example

### LINKED ISSUE #26352: The example is way too long, and should be split up as individual examples for each value instead
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/preserveAspectRatio

### What specific section or headline is this issue about?

examples

### What information was incorrect, unhelpful, or incomplete?

The example is way too long, and should be split up as individual examples for each value instead

### What did you expect to see?

examples to show what each value does, inlined with the documentation for that value.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

It is currently impossible to tell which image in the "final image" relates to which part of the (incredibly scrolled-off-page) code example.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/svg/attribute/preserveaspectratio`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/preserveAspectRatio
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/svg/attribute/preserveaspectratio/index.md
* Last commit: https://github.com/mdn/content/commit/b4f998244660723175f8e06b5d77f68cfb1d1f1a
* Document last modified: 2023-03-06T23:38:10.000Z

</details>

### PR DESCRIPTION
**Pull request by Jenna59**
### Description

Changes related to issue 26352 Content:SVG. Seperated an extremely large code block into smaller code blocks and created multiple iframes with examples in them, instead of just one iframe.

### Motivation

These changes make the article on preserveAspectRatio more clear to understand and easy to read. They group like content and examples.

### Related issues and pull requests

Fixes #26352

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ yarn-error.log*
 _githistory.json
 .DS_Store
 .idea
+.yarn
```
Comment (wbamberg) [inline .gitignore:9]: I think this should not be included in the PR?

File: files/en-us/web/svg/attribute/preserveaspectratio/index.md [hunk 2]
```diff
@@ -15,182 +15,370 @@ Because the aspect ratio of an SVG image is defined by the {{SVGAttr('viewBox')}
 
 ## Example
 
+### Meet (width > height)
```
Comment (wbamberg) [inline files/en-us/web/svg/attribute/preserveaspectratio/index.md:18]: It's arguable, but I would probably have lowercase "meet" (and "slice"), because it's an identifier.

File: files/en-us/web/svg/attribute/preserveaspectratio/index.md [hunk 3]
```diff
@@ -15,182 +15,370 @@ Because the aspect ratio of an SVG image is defined by the {{SVGAttr('viewBox')}
...
+
+This example shows the use of **`meet`** when the element's **`width`** is greater than its **`height`**. It presents three variations, with three different alignment values: **`xMidYMid`**, **`xMinYMid`**, and **`xMaxYMid`**.
```
Comment (wbamberg) [inline files/en-us/web/svg/attribute/preserveaspectratio/index.md:20]: I would just use code formatting, not bold formatting as well.

File: files/en-us/web/svg/attribute/preserveaspectratio/index.md [hunk 4]
```diff
@@ -15,182 +15,370 @@ Because the aspect ratio of an SVG image is defined by the {{SVGAttr('viewBox')}
...
+  <defs>
+    <path
+      id="smiley"
+      d="M50,10 A40,40,1,1,1,50,90 A40,40,1,1,1,50,10 M30,40 Q36,35,42,40 M58,40 Q64,35,70,40 M30,60 Q50,75,70,60 Q50,75,30,60" />
+  </defs> 
+```
+
+```html
   <!-- none -->
   <rect x="0" y="30" width="160" height="60">
```
Comment (wbamberg) [inline files/en-us/web/svg/attribute/preserveaspectratio/index.md:383]: There is a weird gap at the top of this example. Maybe `y` should be `0` here and at line 392?

File: files/en-us/web/svg/attribute/preserveaspectratio/index.md [hunk 5]
```diff
@@ -15,182 +15,370 @@ Because the aspect ratio of an SVG image is defined by the {{SVGAttr('viewBox')}
...
       id="smiley"
       d="M50,10 A40,40,1,1,1,50,90 A40,40,1,1,1,50,10 M30,40 Q36,35,42,40 M58,40 Q64,35,70,40 M30,60 Q50,75,70,60 Q50,75,30,60" />
-  </defs>
+  </defs> 
+```
 
-  <!-- (width>height) meet -->
-  <rect x="0" y="0" width="20" height="10">
+```html
+  <rect x="0" y="0" width="60" height="30">
```
Comment (wbamberg) [inline files/en-us/web/svg/attribute/preserveaspectratio/index.md:40]: Maybe consider making these a bit narrower, like 50 - as it is, the example is cut off with a scrollbar on all but the widest content width? Just a suggestion.

**Review by wbamberg — COMMENTED** [review]
This looks great to me. I had a few really minor comments.

**Comment by Jenna59** [comment]
> This looks great to me. I had a few really minor comments.

If I make the changes, will I have to submit a completely new PR?

**Comment by wbamberg** [comment]
> > This looks great to me. I had a few really minor comments.
> 
> @wbamberg If I make the changes, will I have to submit a completely new PR? 

No.

> Or do I just make the changes, push them, & click Resolve Conversation? 

Exactly, this. Except I don't mind if you click Resolve Conversation.

> I'm on the fence about the meet vs. Meet, too. I'll probably just change it and live with it. lol

I'm happy with whichever you prefer here.

File: files/en-us/web/svg/attribute/preserveaspectratio/index.md (hunk 5 above)
Comment (Jenna59) [reply files/en-us/web/svg/attribute/preserveaspectratio/index.md:40]: Added display: flex to the body of each iframe for responsivity to screen size.

**Comment by Jenna59** [comment]
> I'm happy with whichever you prefer here.

I just pushed. I believe something is being blocked because I need a maintainer to review again.

**Comment by wbamberg** [comment]
I think you need to revert the change to yarn.lock here, so the PR only contains the change to the preserveAspectRatio page. Let me know if you want help trying to do that, and I'll give it a go :).

File: .gitignore [hunk 6]
```diff
@@ -6,3 +6,4 @@ yarn-error.log*
 _githistory.json
 .DS_Store
 .idea
+
```
Comment (caugner) [inline .gitignore:9]:
```suggestion
```

**Comment by Jenna59** [comment]
> I think you need to revert the change to yarn.lock here, so the PR only contains the change to the preserveAspectRatio page. Let me know if you want help trying to do that, and I'll give it a go :).

@wbamberg I'm not 100% sure. I had a merge conflict with the yarn.lock file, so to solve the conflict, I accepted the changes from MDN/content to put it back into alignment with the upstream branch. @caugner seems to be suggesting I had an unnecessary blank space in the .gitignore. I'm wondering if the linting error has thrown everything?

**Comment by wbamberg** [comment]
I'm not sure either, but i do think the yarn.lock changes in here are the cause of the failure. And since your PR doesn't change the mdn/content dependencies, it should not include and changes to yarn.lock.

**Comment by Jenna59** [comment]
I think maybe the yarn.lock file might need to be removed and reinstalled? https://gist.github.com/kenmori/2b31d0cde603751cf92afe2e2bb20263

File: files/en-us/web/svg/attribute/preserveaspectratio/index.md (hunk 5 above)
Comment (wbamberg) [reply files/en-us/web/svg/attribute/preserveaspectratio/index.md:45]: ^this is a much better fix than my suggestion

File: files/en-us/web/svg/attribute/preserveaspectratio/index.md [hunk 7]
```diff
@@ -11,188 +11,401 @@ spec-urls:
...
+}
+
+
+```
+
+{{EmbedLiveSample('meet width height', '100%', 200)}}
+
+### slice (width > height)
+
+This example shows the use of `slice` when the element's **`width`** is greater than its `height`. It presents three variations, with three different alignment values: `xMidYMin`, `xMidYMid`, and `xMidYMax`.
```
Comment (wbamberg) [inline files/en-us/web/svg/attribute/preserveaspectratio/index.md:112]:
```suggestion
This example shows the use of `slice` when the element's `width` is greater than its `height`. It presents three variations, with three different alignment values: `xMidYMin`, `xMidYMid`, and `xMidYMax`.
```

**Review by wbamberg — APPROVED** [review]
👍 looks great, thank you @Jenna59 and congrats on getting your first PR to mdn/content merged 🎉 !
